src/cmap/commands.ts

```diff
@@ -781,11 +810,11 @@ class BinMsg {
     }
 
     if (this.documents.length === 1 && documentsReturnedIn != null && raw) {
-      const fieldsAsRaw: any = {};
+      const fieldsAsRaw: Document = {};
       fieldsAsRaw[documentsReturnedIn] = true;
       _options.fieldsAsRaw = fieldsAsRaw;
 
-      const doc = BSON.deserialize(this.documents[0], _options);
+      const doc = BSON.deserialize((this.documents[0] as unknown) as Buffer, _options);
```

Comment:
@nbbeeken still looking for an answer on this one

Reply:
Whoops missed this, I made documents have type `(Buffer | Document)[]` because it is either depending on the raw flag, it requires casting still but cleaner than this at least. I could add runtime type checks? but there's not really a clear path to how to error, since they should always be translated to Docs when we access these properties as such.